lib/rules/media-query-no-invalid/index.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const {
+	isMediaQueryInvalid,
+	isGeneralEnclosed,
+	isMediaFeaturePlain,
+	isMediaFeatureRange,
+	isMediaFeatureBoolean,
+} = require('@csstools/media-query-list-parser');
+
+const atRuleParamIndex = require('../../utils/atRuleParamIndex');
+const parseMediaQuery = require('../../utils/parseMediaQuery');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+const isCustomMediaQuery = require('../../utils/isCustomMediaQuery');
+const { rangeTypeMediaFeatureNames } = require('../../reference/mediaFeatures');
+
+const ruleName = 'media-query-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (query) => `Unexpected invalid media query "${query}"`,
+});
+
+const HAS_MIN_MAX_PREFIX = /^(?:min|max)-/i;
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/media-query-no-invalid',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules(/^media$/i, (atRule) => {
+			/** @type {Array<{tokens(): Array<import('@csstools/css-tokenizer').CSSToken>}>} */
+			let invalidNodes = [];
+
+			parseMediaQuery(atRule).forEach((mediaQuery) => {
+				if (isMediaQueryInvalid(mediaQuery)) {
+					// Queries that fail to parse are invalid.
+					invalidNodes.push(mediaQuery);
+
+					return;
+				}
+
+				mediaQuery.walk((entry) => {
+					// All general enclosured nodes are invalid.
+					if (isGeneralEnclosed(entry.node)) {
```

Comment:
minor

add `const node = entry.node;`
it's being accessed 8 times

Reply:
Thank you, I will keep this in mind!